WcaOnRails/app/models/user.rb

```diff
@@ -442,19 +442,18 @@ def cannot_edit_data_reason_html(user_to_edit)
     # Don't allow editing data if they have a WCA ID, or if they
     # have already registered for a competition. We do allow admins and delegates
     # who have registered for a competition to edit their own data.
-    if user_to_edit.wca_id
-      # Not using _html suffix as automatic html_safe is available only from
-      # the view helper
-      return I18n.t('users.edit.cannot_edit.msg',
-                    reason: I18n.t('users.edit.cannot_edit.reason.assigned'),
-                    delegate_url: Rails.application.routes.url_helpers.delegates_path).html_safe
-    end
-    if user_to_edit == self && !(admin? || any_kind_of_delegate?) && user_to_edit.registrations.count > 0
-      return I18n.t('users.edit.cannot_edit.msg',
-                    reason: I18n.t('users.edit.cannot_edit.reason.registered'),
-                    delegate_url: Rails.application.routes.url_helpers.delegates_path).html_safe
-    end
-    return nil
+    cannot_edit_reason = if user_to_edit.wca_id
+                           # Not using _html suffix as automatic html_safe is available only from
+                           # the view helper
+                           I18n.t('users.edit.cannot_edit.reason.assigned')
+                         elsif user_to_edit == self && !(admin? || any_kind_of_delegate?) && user_to_edit.registrations.count > 0
+                           I18n.t('users.edit.cannot_edit.reason.registered')
+                         else
+                           nil
+                         end
+    I18n.t('users.edit.cannot_edit.msg',
+           reason: cannot_edit_reason,
+           delegate_url: Rails.application.routes.url_helpers.delegates_path).html_safe if cannot_edit_reason
```

Comment:
If you split something into multilple lines it's worth putting the ` if ... end` rather than this short version.